include/flatbuffers/reflection.h

```diff
@@ -243,7 +255,28 @@ template<typename T> T *GetAnyFieldAddressOf(const Struct &st,
 template<typename T> bool SetField(Table *table, const reflection::Field &field,
                                    T val) {
   assert(sizeof(T) == GetTypeSize(field.type()->base_type()));
-  return table->SetField(field.offset(), val);
+  T def;
+  switch (field.type()->base_type()) {
```

Comment:
That would work yeah, but are we worried about that being brittle if the enum values change? If that's something we don't care about, I'm okay with either implementation. Imho the switch makes the intent more clear, but nbd either way.